modules/mini-games/tightspot.lua

```diff
@@ -0,0 +1,724 @@
+    for i, entity in ipairs(save.entities) do
+        local name = entity.name
+        if name ~= "character" then
+            local table = {name,entity.position, entity.force, entity.minable}
+            save.entities[i] = table
+        else
+            if i == #save.entities then
+                save.entities[i] = nil
+            else
+                local ent = save.entities[#save.entities]
+                name = ent.name
+                local table = {name,ent.position, ent.force, ent.minable}
+                save.entities[i] = table
+                save.entities[#save.entities] = nil
+            end
+        end
+    end
+end
+
+local function market_setup()
+    for i, pos in ipairs(markets)  do
+        local market = variables["surface"].find_entity("market", pos)
+        market.clear_market_items()
+        for _, item in ipairs(variables.level.items) do
+            local offer = {price = {{"coin", tightspot_prices[item]}}, offer = {type = "give-item", item = item}}
+            market.add_market_item(offer)
+        end
+    end
+end
+local function start(args)
+    variables["level"] = {}
+    variables["surface"] = {}
+    variables["walls"] = {}
+    local level_index = args[1]
+    variables.level = config[level_index]
+    variables.diffuclty = variables.level.money[args[2]]
+    variables.loan_price =  variables.level.loan_prices[args[2]]
+    variables["surface"] = game.surfaces[variables.level["surface"]]
+    if not save["tiles"][1] then
+        level_save()
+    end
+    for i, player in ipairs(game.connected_players) do
+        player_join_game(player,i-1)
+    end
+    local force = game.players[1].force
+    force.disable_all_prototypes()
+    local recipe_list = force.recipes
+    for index, item in pairs(variables.level.recipes) do
+        recipe_list[item].enabled = true
+    end
+
+    market_setup()
+    variables.tick = game.tick
+end
+
+
+local function reset_table(table)
+    for i, _ in pairs(table) do
+        table[i] = nil
+    end
+end
+
+local function getSuffix (n)
+    local lastTwo, lastOne = n % 100, n % 10
+    if lastTwo > 3 and lastTwo < 21 then return "th" end
+    if lastOne == 1 then return "st" end
+    if lastOne == 2 then return "nd" end
+    if lastOne == 3 then return "rd" end
+    return "th"
+end
+
+local function Nth (n) return n..getSuffix(n) end
+
+
+
+local function stop()
+    game.speed = 1
+    for i,player in ipairs(game.connected_players) do
+        player.set_controller {type = defines.controllers.god}
+        player.create_character()
+        --gui
+        local Main_gui = Gui.get_left_element(player, game_gui)
+        local table = Main_gui.container["slider"].table
+        table.visible = false
+        Gui.toggle_left_element(player,  game_gui, false)
+    end
+    started[1] = false
+
+    local area = variables.level.area
+    area[1][1] = area[1][1]
+    area[2][1] = area[2][1]
```

Comment:
Does nothing.

Reply:
this 100% needed (jk) lol i have no idea why that is their